trace/config.go

```diff
@@ -28,20 +28,20 @@ type TracerConfig struct {
 }
 
 // InstrumentationVersion returns the version of the library providing instrumentation.
-func (t *TracerConfig) InstrumentationVersion() string {
+func (t TracerConfig) InstrumentationVersion() string {
 	return t.instrumentationVersion
 }
 
 // SchemaURL returns the Schema URL of the telemetry emitted by the Tracer.
-func (t *TracerConfig) SchemaURL() string {
+func (t TracerConfig) SchemaURL() string {
 	return t.schemaURL
 }
 
 // NewTracerConfig applies all the options to a returned TracerConfig.
-func NewTracerConfig(options ...TracerOption) *TracerConfig {
-	config := new(TracerConfig)
+func NewTracerConfig(options ...TracerOption) TracerConfig {
+	var config TracerConfig
 	for _, option := range options {
-		option.apply(config)
+		option.apply(&config)
```

Comment:
Should this be updated to the alternate function signature, it takes and returns a struct instead of a pointer?